controllers/dscinitialization/servicemesh_setup.go

```diff
@@ -7,30 +7,37 @@ import (
 	corev1 "k8s.io/api/core/v1"
 
 	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	infrav1 "github.com/opendatahub-io/opendatahub-operator/v2/infrastructure/v1"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/servicemesh"
 )
 
 func (r *DSCInitializationReconciler) configureServiceMesh(instance *dsciv1.DSCInitialization) error {
-	switch instance.Spec.ServiceMesh.ManagementState {
-	case operatorv1.Managed:
-		serviceMeshFeatures := feature.ClusterFeaturesHandler(instance, configureServiceMeshFeatures())
-
-		if err := serviceMeshFeatures.Apply(); err != nil {
-			r.Log.Error(err, "failed applying service mesh resources")
-			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed applying service mesh resources")
-			return err
-		}
-	case operatorv1.Unmanaged:
-		r.Log.Info("ServiceMesh CR is not configured by the operator, we won't do anything")
-	case operatorv1.Removed:
-		r.Log.Info("existing ServiceMesh CR (owned by operator) will be removed")
-		if err := r.removeServiceMesh(instance); err != nil {
-			return err
+	emptySpec := infrav1.ServiceMeshSpec{}
+	if instance.Spec.ServiceMesh != emptySpec {
+		switch instance.Spec.ServiceMesh.ManagementState {
+		case operatorv1.Managed:
+			serviceMeshFeatures := feature.ClusterFeaturesHandler(instance, configureServiceMeshFeatures())
+
+			if err := serviceMeshFeatures.Apply(); err != nil {
+				r.Log.Error(err, "failed applying service mesh resources")
+				r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed applying service mesh resources")
+				return err
```

Comment:
```suggestion
				r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed applying service mesh resources")
				return fmt.Errorf(""failed applying service mesh resources: %w", err)
```

but also you need to import `"fmt"`